### ISSUE TITLE
Why is it disallowed to change storageConfig (ie, increase disk size?)

### ISSUE THREAD

**Issue by CarlQLange** [datastax/cass-operator#390]
I see that it is disallowed to change the storage spec of a DC: https://github.com/datastax/cass-operator/blob/71806eaa1c30e8555abeca9f4db12490e8658e1a/operator/pkg/apis/cassandra/v1beta1/webhook.go#L109-L111

Unfortunately I found this out as I ran out of space in my DC. Before I used cass-operator, I simply resized the PVCs without any issue. Is there no way to do this with cass-operator? I would have thought it was mildly common for DCs to run out of space and need increasing.

**Comment by CarlQLange** [comment]
Separately it would be nice to have disk failure handled at an operator level, although I don't know what that behaviour would look like.

**Comment by jimdickinson** [comment]
Hi @CarlQLange - we put in validation and logic to block this because StatefulSet also disables mutating the volume claim templates on an existing statefulset. There are workarounds, however, and this is a popular request. We'll try to write up a tutorial that covers some of them, but it's very "at your own risk". One of the least risky options if you're on the cloud is using some off-the-shelf docs for how to resize a volume from your cloud provider. The volume will be out of sync with the `CassandraDatacenter` / `StatefulSet` / `PersistentVolumeClaim`, and if you later want to add Cassandra nodes by increasing the `size` on your `cassdc` the new volume will still be the original size you started with, but you can get more space for now and you'll be unblocked.

**Comment by CarlQLange** [comment]
Hey, thanks a lot for answering on the weekend! I did resize the disks, but for some reason the cassandra pods think they're still the original size. However, that's almost certainly my own fault rather than the fault of the operator.

I think it would be great to hear some other workarounds. I was wondering about adding a new DC and migrating the data from the old one, for example, but that sounds scary. Really appreciate your taking the time out to have responded, thank you.

**Comment by CarlQLange** [comment]
I note that in my case, this issue is compounded by Azure not allowing resizing of disks without detaching them first. Unfortunately I can't kill a cassandra pod and have it stay dead while I issue the resize, of course...
